src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -100,6 +100,10 @@ private function generateMatchMethod($supportsRedirections)
     {
         $code = rtrim($this->compileRoutes($this->getRoutes(), $supportsRedirections), "\n");
 
+        if ('' === $code && class_exists('Symfony\Bundle\FrameworkBundle\Controller\WelcomeController')) {
```

Comment:
This would introduce a big implementation detail in the Routing component. I don't think it's acceptable to do this, also because you need to detect if you are in debugging mode (which is currently not checked?).